app/unit-tests/src/org/commcare/CommCareTestApplication.java

```diff
@@ -66,6 +67,9 @@ public class CommCareTestApplication extends CommCareApplication implements Test
 
     @Override
     public void onCreate() {
+        // set if before calling super to initialte the dataChangeLogger correctly
```

Comment:
@shubham1g5 are these changes motivated/required by the Robolectric update?

Reply:
Yeah after the update the data change log test started failing [over here](https://github.com/dimagi/commcare-android/blob/master/app/unit-tests/src/org/commcare/logging/DataChangeLoggerTest.java#L28) as the application onCreate() was no longer getting called before [setup function](https://github.com/dimagi/commcare-android/blob/master/app/unit-tests/src/org/commcare/logging/DataChangeLoggerTest.java#L23) where we were earlier initialising the logger. I didn't go deep into what aspect of robolectric update caused this, but instead tried to fix the [original data change logger init](https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/CommCareApplication.java#L208) as part of application onCreate() by setting the storage here.